### PR TITLE
Mysql

### PR DESCRIPTION
**Pull request by jonathan-s**
Merging the MySQL plugin into the sd-agent-plugins.

### CODE REVIEW

File: MySQL/readme.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+*  **Connection Usage** - This is the current connection count (`threads running`) as a percentage of `max connections`. If you are close to 100% you have an immense load or you might consider to increase the `max connections`.
+*  **Key cache hit ratio** - This is the proportion of index values that are read from cache instead of from disk. 
+*  **Reads/s** - This is a calculation of the number of reads that's done on the MySQL server. It's a combination of `Com_select`, `Qcache_hits`.
+*  **Writes/s** - This is a calculation of the number of writes that's done on the MySQL server. It's a combination of `Com_insert`, `Com_replace`, `Com_update`, `Com_delete`.
+*  **RW ratio** - This is the Read and Write ratio. This is simply the ratio of reads and writes as how reads and writes are defined above.
+*  **Tmp Cache Hit Ratio** - This is the proportion of temp tables that were created in cache or on disk. If the value is 100% it means that all tables were created in cache.
+
+For a more in-depth guide of understanding what these metrics mean please check out the blog post on [How to monitor MySQL](#)
```
Comment (dmytton) [inline MySQL/readme.md]: Has this been written?

**Comment by jonathan-s** [comment]
What do you mean, has this been written? If I wrote it? If that's the question the answer is yes.

**Comment by dmytton** [comment]
It's a line note on line 146 of the readme - has the blog post been written?

**Comment by jonathan-s** [comment]
No, the blog post itself has not been written yet. I didn't have time to do that yet.

**Comment by dmytton** [comment]
You have some PEP8 errors to resolve - just looks like `E501 line too long` errors, plus 1 `E302 expected 2 blank lines, found 1`

**Comment by bassdread** [comment]
Version should be 1.0 :)

**Comment by dkerwin** [comment]
Will this be merged any time soon?

File: MySQL/mysql.py [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+            try:
+                status['qcache hits'] = self.get_db_results(
+                    db, 'SHOW STATUS LIKE "Qcache_hits"')
+
+                qcache_ps = self.calculate_per_s('qcache_ps', status[
+                    'qcache hits'])
+                status['qcache hits/s'] = qcache_ps
+
+                select = self.get_db_results(
+                    db, 'SHOW STATUS LIKE "Com_select')
```
Comment (dkerwin) [inline MySQL/mysql.py]: Missing " after Com_select

File: MySQL/mysql.py [hunk 3]
```diff
@@ -0,0 +1,663 @@
...
+            # Replication - Seconds Behind Master
+            secondsBehindMaster = None
+            try:
+                cursor = db.cursor(MySQLdb.cursors.DictCursor)
+                cursor.execute('SHOW SLAVE STATUS')
+                result = cursor.fetchone()
+
+            except MySQLdb.OperationalError as message:
+
+                self.mainLogger.error(
```
Comment (dkerwin) [inline MySQL/mysql.py]: Should be checks_logger or am i missing something? Happens several times...

**Comment by jonathan-s** [comment]
@dkerwin It's been merged now. Thanks for taking the time to look at the code too!

**Pull request merged**